dallinger/prolific.py

```diff
@@ -281,3 +282,98 @@ def _req(self, method: str, endpoint: str, **kw) -> dict:
+                                "status": "status",
+                                "created": "created-date",
+                                "expiration": "",
+                                "description": "",
+                            }
+                        ]
+                    }
+
+            elif method == "POST":
+                # method="POST", endpoint="/studies/", json=payload
+                if endpoint == "/studies/":
+                    response = {
+                        "id": "study-id",
+                        "external_study_url": "external-study-url",
+                    }
+
+                # method="POST", endpoint=f"/studies/{study_id}/transition/", json={"action": "PUBLISH"},
+                elif re.match(r"/studies/[a-z0-9]+/transition/", endpoint):
+                    response = True
+
+            elif method == "PATCH":
+                # method="PATCH", endpoint=f"/studies/{study_id}/", json={"total_available_places": new_total},
+                response = {
+                    "items": ["https://experiment-url-1", "https://experiment-url-2"],
+                    "message": "More info about this particular recruiter's process",
+                }
+
+            # method="DELETE", endpoint=f"/studies/{study_id}"
+            elif method == "DELETE":
+                response = {"status_code": 204}
+
+        # method="POST", endpoint=f"/bulk-bonus-payments/{setup_response['id']}/pay/"
+        elif endpoint.startswith("/bulk-bonus-payments/"):
+            response = {"id": str(uuid4())}
+
+        # method="POST", endpoint="/submissions/bonus-payments/", json=payload
+        elif endpoint.startswith("/submissions/bonus-payments"):
+            response = {"id": str(uuid4())}
+
+        elif endpoint.startswith("/submissions/"):
+            # method="POST", endpoint=f"/submissions/{session_id}/transition/", json={"action": "APPROVE"},
+            if re.match(r"/submissions/[A-Za-z0-9]+/transition/", endpoint):
+                response = True
+
+            # method="GET", endpoint=f"/submissions/{session_id}/"
+            elif re.match(r"/submissions/[A-Za-z0-9]+/", endpoint):
+                response = {
+                    "id": "id",
+                    "study_id": "study-id",
+                    "participant": "participant",
+                    "started_at": "started-at-timestamp",
+                    "status": "AWAITING REVIEW",
+                }
+
+        if response is None:
+            raise RuntimeError("Simulated Prolific API call could not be matched.")
+        self.debug_log_response(response)
+        return response
+
+    def debug_log_request(self, method, endpoint, **kw):
+        log_msg = (
+            f'Simulated Prolific API request: method="{method}", endpoint="{endpoint}"'
+        )
+        log_msg += f', json={kw["json"]}' if "json" in kw else ""
+        log_msg += f'\n{kw["message"]}' if "message" in kw else ""
+        logger.warning(log_msg)
```

Comment:
I think `warning` should be saved for situations where something bad is happening. Better instead to use `info` here.

Reply:
The problem is with `info` that not all log messages get printed. That's why I used `warning`. Maqybe this has to do with the worker log level not being properly configured?

Reply:
I know you had trouble with this previous issue, but I think I just managed to fix it by replacing

```py
    log_level = os.environ.get("loglevel", "WARN")
    logging.basicConfig(
        format="%(asctime)s %(message)s",
        level=getattr(logging, log_level, logging.WARN),
    )
```

with

```py
    log_level = os.environ.get("loglevel", "INFO")
    logging.basicConfig(
        format="%(asctime)s %(message)s",
        level=getattr(logging, log_level, logging.INFO),
    )
```

Reply:
Can you check if it works for you?